### PR TITLE
Add order command

### PR DESCRIPTION
**Pull request by guanquann**
- Add `addorder` command `[addorder [INDEX] r/ [DESCRIPTION] d/ [DATE]]`
- For example: `addorder 1 r/ test d/ 2024-04-01`
- Show the number of orders from each supplier in GUI
- Added most testing
- User guide

image

### CODE REVIEW

**Review by maze508 — COMMENTED** [review]
When I type `addorder 0 r/ something d/ 2024-04-01` it returns:
```
Invalid command format! 
addorder: Adds an order to the person identified by the index number used in the last person listing.
Parameters: INDEX (must be a positive integer) d/ [DATE] r/ [DESCRIPTION] 
Example: addorder 1 d/ 2024-01-01 r/ 100 chicken wings.
```

but when I type `addorder 100 r/ something d/ 2024-04-01` it returns:

```
The person index provided is invalid
```

You should standardise the error messages for similar errors

File: docs/UserGuide.md [hunk 1]
```diff
@@ -131,6 +131,24 @@ Examples:
 * `find alex david` returns `Alex Yeoh`, `David Li`<br>
   ![result for 'find alex david'](images/findAlexDavidResult.png)
 
+### Adding an order : `addorder`
+
+Adds an order to a supplier.
+
+Format: `addorder INDEX d/DATE r/DESCRIPTION`
```
[the remark this answers is not in the document]
Comment (guanquann) [reply docs/UserGuide.md:138]: They technically don't need to be separated 😄

[the remark this answers is not in the document]
Comment (maze508) [reply docs/UserGuide.md:138]: Think I was not clear, I mean you should explicity state something like this `addorder 1 r/n/d/r/d/r/d/2024-02-01` shouldn't work

File: src/main/java/seedu/address/logic/parser/AddOrderCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        }
+
+        return new AddOrderCommand(index, arrivalDate, description);
+    }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
```
[the remark this answers is not in the document]
Comment (guanquann) [reply src/main/java/seedu/address/logic/parser/AddOrderCommandParser.java:61]: Good catch!

File: src/main/java/seedu/address/logic/parser/ArgumentMultimap.java [hunk 3]
```diff
@@ -109,4 +107,12 @@ private void checkValuesAlpha(List<String> values, Prefix prefix) throws ParseEx
             throw new ParseException(String.format(Messages.MESSAGE_ALPHABET_ONLY, prefix.getPrefix()));
         }
     }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    public boolean arePrefixesPresent(Prefix... prefixes) {
```
Comment (maze508) [inline src/main/java/seedu/address/logic/parser/ArgumentMultimap.java:115]: Dupliucate Command. Consider reusing already defined methods

**Pull request merged**